src/controllers/trust.involved.controller.ts

```diff
@@ -0,0 +1,102 @@
+import { NextFunction, Request, Response } from 'express';
+import { TrusteeType } from '../model/trustee.type.model';
+import * as config from '../config';
+import { logger } from '../utils/logger';
+
+const TRUST_INVOLVED_TEXTS = {
+  title: 'Individuals or entities involved in the trust',
+};
+
+const get = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    const templateName = config.TRUST_INVOLVED_PAGE;
+
+    return res.render(
+      templateName,
+      {
+        backLinkUrl: `${config.TRUST_DETAILS_URL}/${req.params['id']}`,
+        templateName,
+        pageData: {
+          trusteeType: TrusteeType,
+        },
+        pageParams: {
+          title: TRUST_INVOLVED_TEXTS.title,
+          checkYourAnswersUrl: config.CHECK_YOUR_ANSWERS_URL,
+        },
+      },
+    );
+  } catch (error) {
+    logger.errorRequest(req, error);
+    next(error);
+  }
+};
+
+const post = (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+    if (req.body.noMoreToAdd) {
+      return res.redirect(config.CHECK_YOUR_ANSWERS_URL);
+    }
+
+    const typeOfTrustee = req.body.typeOfTrustee;
+    // the req.params['id'] is already validated in the has.trust.middleware but sonar can not recognise this.
+    const url = `${config.TRUST_INVOLVED_URL}/${req.params['id']}`;
+
+    switch (typeOfTrustee) {
+        case TrusteeType.HISTORICAL:
+          logger.info("TODO: Route to trust-historical-beneficial-owner page ");
+          if (isValidUrl (url) ) {
+            return res.redirect(url);
+          }
+          break;
+        case TrusteeType.INDIVIDUAL:
+          logger.info("TODO: Route to trust-individual page when story coded ");
+          if (isValidUrl (url) ) {
+            return res.redirect(url);
+          }
+          break;
+        case TrusteeType.LEGAL_ENTITY:
+          logger.info("TODO: Route to trust-ole page when story coded ");
+          if (isValidUrl (url) ) {
+            return res.redirect(url);
+          }
+          break;
+        default:
+          logger.info("TODO: On validation No trustee selected, re-displaying page");
+          if (isValidUrl (url) ) {
+            return res.redirect(url);
+          }
+    }
+
+  } catch (error) {
+    logger.errorRequest(req, error);
+
+    return next(error);
+  }
+};
+
+// Required for Sonar rule tssecurity:S5146 (this will never happen but Sonar can not understand middleware in this case)
+const isValidUrl = (url: string) => {
+  if (url) {
```

Comment:
you still need to check the url starts with an expected string, isValidUrl will return true if it receives http;//hack.me

Reply:
Just trying to find the simplest way to fool sonar and failed

I will do a check and add some more comments.

I don't like the fact that we have to do extra code just to remove a Sonar error that is NOT an error

I will slack you when the sonar error is gone

Reply:
I think Sonar is correct, we are using the post function in one scenario (in a route with middleware) but the post function is available to be called from anywhere in the codebase. If there is the potential for this to happen then we should add the recommended check.

Reply:
I can not imagine that scenario although  being implemented - but it could exist if someone coded it that way.

Still I will do a check after a meeting and let you know it is pushed

Reply:
This rule should be removed
Rule check impossible for parameters scenario